btree/btree.c

```diff
@@ -714,7 +749,7 @@ struct td {
 	 * read-only after the tree root is loaded into memory.
 	 */
 	struct m0_rwlock            t_lock;
-	struct segaddr              t_root;
+	struct nd                  *t_root;
```

Comment:
Two points: proper node_get() function also needs code to page fault the node asynchronously (from another thread). That requires some additional discussion. I think that for the time being we should continue with the mock seg_ops and concentrate on the state machines.

Also, I was wrong about the root node: it is never relocated (because it can be embedded in a larger ambient structure, for one thing), so keeping a pointer to its node descriptor is fine, provided m0_btree_open() pins the root node descriptor in memory and m0_btree_close() releases it. But this moves the complexity to m0_btree_open(): it now has to be a non-blocking state machine that loads the root node from the memory.